nuxeo-compound-documents/src/main/java/org/nuxeo/compound/documents/CompoundDocumentServiceImpl.java

```diff
@@ -106,7 +109,19 @@ protected void createEntry(DocumentModel compoundDoc, ZipFile zip, ZipEntry entr
             var doc = session.createDocumentModel(parentDocPath, entryDocName, compoundFolderType);
             session.createDocument(doc);
         } else {
-            var blob = new ZipEntryBlob(zip, entry);
+            Blob blob;
+            if (entry.getName().endsWith(".zip")) {
```

Comment:
I think we can make something more resilient, like a regex so case is not an issue.